src/test/integrationTests/userController.test.js

```diff
@@ -143,4 +143,137 @@ describe('Integration tests for the user controller', () => {
         .to.equal('Invalid request. All fields are required');
     });
   });
+  before('Get user token', async () => {
+    const response = await chai.request(app).post('/api/v1/auth/login')
+      .send({
+        email: 'demo1@demo.com',
+        password: 'password',
+      });
+    const { token, id } = response.body.data.userDetails;
+    describe('Test profile update endpoints', () => {
+      const newData = {
+        firstName: 'Kelechi',
+        lastName: 'Janet',
+        password: 'ejie565ewiw',
+        birthdate: '1/5/2018',
+        gender: 'female',
+        preferredLanguage: 'English',
+        preferredCurrency: 'USD',
+        profileImage: 'https://profileurl.com',
+        city: 'Enugu',
+        state: 'Enugu',
+        zip: '154587',
+        country: 'Nigeria',
+        role: 'SuperAdmin',
+        department: 'Accounting',
+        lineManager: 'Daniels',
+      };
+      it('should update users profile', async () => {
+        const result = await chai.request(app).patch(`/api/v1/profile/${id}`)
+          .set('x-access-token', token)
+          .send(newData);
+        expect(result.body.data.success).to.equal(true);
+        expect(result.body.data.userDetails.firstName).to
+          .equal(newData.firstName);
+        expect(result.body.data.userDetails.lastName).to
+          .equal(newData.lastName);
+        expect(result.body.data.userDetails.preferredLanguage).to
+          .equal(newData.preferredLanguage);
+        expect(result.body.data.userDetails.preferredCurrency).to
+          .equal(newData.preferredCurrency);
+        expect(result.body.data.userDetails.city).to
+          .equal(newData.city);
+        expect(result.body.data.userDetails.state).to
+          .equal(newData.state);
+        expect(result.body.data.userDetails.zip).to
+          .equal(newData.zip);
+        expect(result.body.data.userDetails.country).to
+          .equal(newData.country);
+        expect(result.body.data.userDetails.role).to
+          .equal(newData.role);
+        expect(result.body.data.userDetails.department).to
+          .equal(newData.department);
+        expect(result.body.data.userDetails.lineManager).to
+          .equal(newData.lineManager);
+      });
```

Comment:
It would also prevent this long list of assertions. you can only test to update two or three columns or even just one and it would suffice.